# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/base/strings/string_piece.cc
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/base/strings/string_piece.cc
--- a/security/sandbox/chromium/base/strings/string_piece.cc	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/base/strings/string_piece.cc	Wed Jun 12 11:10:48 2019 +0100
@@ -11,6 +11,7 @@
 #include <ostream>
 
 #include "base/logging.h"
+#include "base/strings/utf_string_conversions.h"
 
 namespace base {
 namespace {
@@ -40,18 +41,15 @@
 template class BasicStringPiece<string16>;
 #endif
 
-bool operator==(const StringPiece& x, const StringPiece& y) {
-  if (x.size() != y.size())
-    return false;
-
-  return StringPiece::wordmemcmp(x.data(), y.data(), x.size()) == 0;
-}
-
 std::ostream& operator<<(std::ostream& o, const StringPiece& piece) {
   o.write(piece.data(), static_cast<std::streamsize>(piece.size()));
   return o;
 }
 
+std::ostream& operator<<(std::ostream& o, const StringPiece16& piece) {
+  return o << UTF16ToUTF8(piece);
+}
+
 namespace internal {
 
 template<typename STR>
@@ -261,8 +259,8 @@
 
   for (size_t self_i = pos; self_i < self.size(); ++self_i) {
     bool found = false;
-    for (size_t s_i = 0; s_i < s.size(); ++s_i) {
-      if (self[self_i] == s[s_i]) {
+    for (auto c : s) {
+      if (self[self_i] == c) {
         found = true;
         break;
       }
@@ -329,8 +327,8 @@
 
   for (size_t self_i = std::min(pos, self.size() - 1); ;
        --self_i) {
-    for (size_t s_i = 0; s_i < s.size(); s_i++) {
-      if (self.data()[self_i] == s[s_i])
+    for (auto c : s) {
+      if (self.data()[self_i] == c)
         return self_i;
     }
     if (self_i == 0)
@@ -374,8 +372,8 @@
 
   for (size_t self_i = std::min(pos, self.size() - 1); ; --self_i) {
     bool found = false;
-    for (size_t s_i = 0; s_i < s.size(); s_i++) {
-      if (self.data()[self_i] == s[s_i]) {
+    for (auto c : s) {
+      if (self.data()[self_i] == c) {
         found = true;
         break;
       }